services/r2ctl/server/server.go

```diff
@@ -24,18 +24,23 @@ import (
 	"net/http"
 
 	"github.com/gorilla/mux"
+	"github.com/m3db/m3ctl/handler"
 )
 
-// NewServer creates a new http server
-func NewServer(address string, opts Options) (*http.Server, error) {
-	router := mux.NewRouter()
-	if err := registerHandlers(router, opts.InstrumentOptions()); err != nil {
-		return nil, err
+func registerHandlers(router *mux.Router, controllers []handler.Controller) {
+	for _, c := range controllers {
+		c.RegisterHandlers(router)
 	}
+}
+
+// NewServer creates a new http server for R2
+func NewServer(address string, serverOpts Options, controllers []handler.Controller) *http.Server {
```

Comment:
I'm missing some context, could you clarify the difference between `controller`, `handler`, and `operator`.

Reply:
Handler is just logical grouping of routes. The directory can just as easily be called Controllers.

A Controller is a wrapper around a set of http handlers that registers them with the server and provides some useful state. State can be instrumentation or a path prefix to expose them under. In the case of r2, it also has an implementation of an API to call the functions on. 

An Operator is the thing that actually performs the operations for r2. Each r2 api instance has an operator associated with it. For starters, I'm implementing a stub operator that won't actually talk to etcd but instead will return dummy results. When I implement the real Operator we'll just have to swap out the stub for that one.

Reply:
talked offline, some renaming:

- `API` -> `Handlers`
- `Operator` -> `Service`
- `Controller` -> `Router`
- `Handler` -> folder goes away, separate folders for each router